### PR TITLE
[Added] [BEL-3906] Ruby - was added support to get base url given an environment name

### PR DESCRIPTION
**Pull request by benitocas88**
:question: What
---

- It's possible pass an environment name to `Client` to get base url

`client = Belvo::Client.new('secret-id', 'secret-password', 'sandbox|production|development')`

:speech_balloon: Why
---

- It's easier and friendly pass an environment name avoiding an extra path for base url

 
https://belvoteam.atlassian.net/browse/BEL-3906

### CODE REVIEW

File: lib/belvo.rb [hunk 1]
```diff
@@ -18,6 +19,7 @@ class Client
     # @return [APISession] Authenticated Belvo API session
     def initialize(secret_key_id, secret_key_password, url = nil)
       (belvo_api_url = url) || ENV['BELVO_API_URL']
+      belvo_api_url = Environment.get_url(belvo_api_url)
       raise BelvoAPIError, 'You need to provide a URL.' if belvo_api_url.nil?
```
Comment (anisaoshafi) [inline lib/belvo.rb:23]: Remember to keep the error messages across sdk-s consistent.

Comment (benitocas88) [reply lib/belvo.rb:23]: You right! In belvo-python at the end I used this "You need to provide a URL or a valid environment."  
What do you think about it? @anisaoshafi

Comment (anisaoshafi) [reply lib/belvo.rb:23]: that makes sense to me Beni, I'd go for that one :)

File: spec/belvo/environments_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+RSpec.describe Environment do
+  [
+    %w[sandbox https://sandbox.belvo.com],
+    %w[development https://development.belvo.com],
+    %w[production https://api.belvo.com]
+  ].each do |env|
+    it 'given an environment name should be return a valid api url' do
```
Comment (shekodn) [inline spec/belvo/environments_spec.rb:9]:
```suggestion
    it 'given an environment name should return a valid API URL do
```

File: spec/belvo/environments_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      expect(described_class.get_url(env[0])).to eq(env[1])
+    end
+  end
+
+  it 'given an url should be return the same url as default' do
```
Comment (shekodn) [inline spec/belvo/environments_spec.rb:14]:
```suggestion
  it 'given an URL should return the same URL as default' do
```

File: spec/belvo/client_spec.rb [hunk 4]
```diff
@@ -25,4 +26,31 @@
       described_class.new('foo', 'bar', 'http://fake.api')
     end.not_to raise_error
   end
+
+  [
+    %w[sandbox https://sandbox.belvo.com],
+    %w[development https://development.belvo.com],
+    %w[production https://api.belvo.com]
+  ].each do |env|
+    it 'given an environment the session should be have the correct api url' do
```
Comment (shekodn) [inline spec/belvo/client_spec.rb:35]:
```suggestion
    it 'given an environment the session should have the correct API URL do
```

File: spec/belvo/client_spec.rb [hunk 5]
```diff
@@ -25,4 +26,31 @@
...
+      url = format('%<url>s/api/', url: env[1])
+
+      WebMock.stub_request(:get, url).to_return(status: 200)
+
+      client = described_class.new('key-id', 'key-password', name)
+      expect(client.session.instance_variable_get(:@url)).to eq(url)
+    end
+  end
+
+  it 'given an url the session should be have the same url' do
```
Comment (shekodn) [inline spec/belvo/client_spec.rb:46]:
```suggestion
  it 'given an URL the session should have the same one do
```

Comment (shekodn) [inline spec/belvo/client_spec.rb:46]:
```suggestion
  it 'given an URL the session should have the same one' do
```

**Review by shekodn — COMMENTED** [review]
Left some suggestions. Overall it looks good to me 👍

**Review by shekodn — APPROVED** [review]

**Review by gingrassia — APPROVED** [review]

**Pull request merged**